src/pages/Search/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+import { Paragraph } from '@/components/common/typography/Paragraph';
+import { Text } from '@/components/common/typography/Text';
+import SearchBar from '@/components/common/SearchBar';
+
+export default function SearchPage() {
+  const location = useLocation();
+
+  const query = new URLSearchParams(location.search).get('query');
```

Comment:
```suggestion
  const [searchParams] = useSearchParams();
  const query = searchParams.get('query');
```

react-router-dom에서 기본적으로 제공하는 [useSearchParams](https://reactrouter.com/en/main/hooks/use-search-params#usesearchparams) 훅이 있습니다.